### PR TITLE
[feat] Add DiffusionNFT Wan video RL training

### PR DESCRIPTION
**Pull request by Abecid**
## Purpose

Adds a reproducible DiffusionNFT video RL training path for Wan 2.1 T2V using VideoAlign/VideoReward VQ, MQ, and TA rewards. The PR includes the tracked config, asset preparation script, reward adapters, Modal launcher, docs, and focused local tests needed to reproduce the run without private launch glue.

## Changes

- Add `diffusion_nft_videoalign.yaml` for Wan 2.1 T2V DiffusionNFT training with VideoAlign VQ/MQ/TA rewards, FlowUniPC sampling, CFG guidance, EMA validation, and video-sized latent settings.
- Extend the DiffusionNFT RL method with managed sample-score-train behavior, validation sampling/logging, old-policy checkpoint state, EMA state, reward diagnostics, and corrected gradient accumulation/micro-step handling.
- Add reusable RL reward composition/media helpers plus VideoAlign reward scorers for `videoalign_vq`, `videoalign_mq`, and `videoalign_ta`.
- Keep HPSv3 only as an optional reward scorer (`hpsv3_general`, `hpsv3_percentile`) for future reward presets; no HPSv3 trainer/training-loop files are included.
- Vendor the inference runtime files required for VideoAlign/VideoReward and reward-only HPSv3 scoring under `fastvideo/third_party/rl_rewards/`.
- Add `prepare_diffusion_nft_assets.py` and `diffusion_nft_wan_video.md` so prompt download/preprocessing, reward checkpoint setup, generated run config, and launch steps are reproducible outside private launchers.
- Add `modal_train_diffusion_nft.py` for the tracked Modal training path.
- Fix text-only preprocessing to request padded text encoder outputs.
- Update focused local tests for RL sampling config, CFG sampling, validation helpers, reward weighting, nested reward config parsing, and the video DiffusionNFT YAML layout.

## Cleanup Notes

- Removed unrelated OCR reward adapter code from the PR.
- Removed HPSv3 trainer/training surfaces; retained only reward-inference files needed for optional HPSv3 scoring.
- No DeepSpeed config files are included in the current changed-file list.

## Verification

- Ran a 5-step Modal smoke/probe from this PR path:

```bash
conda run --no-capture-output -n fastvideo modal run --timestamps modal_train_diffusion_nft.py::train --max-train-steps 5
```

Result: completed successfully on retry and synced W&B run `adamlee00/diffusion_nft_wan/xg3v3v9g` with 5 train steps, 16 validation media files, and checkpoint `checkpoint-5`.

- Confirmed the Modal run resolved the same VideoAlign reward map as the July 2 reference run:

```text
reward_backend=genrl
reward_map={'videoalign_vq': 1.0, 'videoalign_mq': 1.0, 'videoalign_ta': 1.0}
```

- Ran `git diff --check`: passed.
- Ran syntax compilation over changed Python/test files with `conda run -n fastvideo python -m compileall ...`: passed.

Local `pytest` and `pre-commit` could not be run in this shell because the local `fastvideo` conda env is missing `pytest`, `torch`, and `PyYAML`, and `pre-commit` is not installed. CI should be treated as the source of truth for those checks.

## Checklist

- [ ] I ran `pre-commit run --all-files` and fixed all issues
- [x] I added or updated tests for my changes
- [x] I updated documentation if needed
- [x] I considered GPU memory impact of my changes

**For model/pipeline changes, also check:**
- [ ] I verified SSIM regression tests pass
- [ ] I updated the support matrix if adding a new model

### CODE REVIEW

**Comment by SolitaryThinker** [comment]
Rebased onto `cae8fa18` to pick up current main/CI dependency fixes; range-diff preserves every non-conflicting commit, with only the net-zero legacy tracker add/remove omitted and main’s deleted `.agents` experiment journal kept deleted.

**Comment by SolitaryThinker** [comment]
Added formatter-only commit `1c91e31bd`; the exact project hook chain now passes on all five previously rewritten files.

**Review by SolitaryThinker — CHANGES_REQUESTED** [review]
I pushed `5ac2cd9f4` to resolve the deterministic review failures I could fix without changing the intended reward stack:

- preserve explicit zero CFG;
- keep MatrixGame validation latents, image conditioning, and actions on the same temporal length;
- disambiguate canonical short-video axes;
- feed the fp32 HPSv3 reward head fp32 activations;
- include the HPSv3 config and vendored licenses in built wheels;
- make the checked-in K-repeat config internally valid;
- carry the documented VideoAlign checkpoint path into the training shell; and
- fail early for unsupported SP and multi-GPU preprocessing instead of silently corrupting samples/files.

Verification at this head: changed-file pre-commit passed, `bash -n` passed, Python compilation passed, both K-repeat divisibility invariants passed, and a clean wheel contains `HPSv3_7B.yaml` plus both vendored licenses. Local pytest cannot collect because this machine's Python environment has no Torch; the fresh CI run is the runtime proof available here.

One P1 blocker remains: the documented launch environment is internally inconsistent. `examples/train/requirements-diffusion-nft.txt:6,11` pins `datasets==3.6.0` and `transformers==4.57.3`, but `pyproject.toml:26,82` requires `transformers>=5.0.0` and `datasets>=4.0.0`. The docs and launchers install both into one environment. Modal additionally installs the Torch 2.12 project and then a FlashAttention wheel explicitly built for Torch 2.10 (`modal_train_diffusion_nft.py:99-108`). Import-only checks do not prove ABI or reward-model compatibility.

Please make the reward runtime compatible with FastVideo's declared dependency set (or otherwise provide one non-conflicting in-process environment), use an attention artifact built for the installed Torch version, and show `pip check` plus the reward preflight in that exact final environment.

**Comment by SolitaryThinker** [comment]
Pushed maintainer fix 518aeab0b to close the current review blockers:

- removed the conflicting reward requirements layer and aligned Modal with the repository CUDA 13.0 / Torch 2.12 / FlashAttention tuple, with dependency and GPU ABI preflights;
- delegated vendored Qwen2-VL reward forwards to the Transformers 5 multimodal model path;
- restored VideoAlign prompt/luminance semantics and fixed FlowUniPC list-sigma dispatch;
- moved the CPU DiffusionNFT suites into the Train Framework CI lane.

Local evidence: 41 focused tests passed on Torch 2.12.1 / Transformers 5.13, pre-commit passed, Python and shell syntax checks passed, and the cu130 resolver selected Torch 2.12.0 + Transformers 5.14. Fresh CI is running in Buildkite 4463. The remaining merge gate is an exact-head Modal reward/runtime smoke; I am keeping the review blocked until that evidence exists.
